homework_3/solution.js

```diff
@@ -1,6 +1,43 @@
-var sum = 0;
+var colors = require("colors");
 
-for (var i = 2; i < process.argv.length; i++) {
-  sum += Number(process.argv[i]);
+var mathFuns = {
+    "+" : function(x, y) { return x + y; },
+    "-" : function(x, y) { return x - y; },
+    "x" : function(x, y) { return x * y; },
+    "/" : function(x, y) { return x / y; },
+    "^" : function(x, y) { return Math.pow(x, y); }
+}
+
+function isNumber(n) {
+  return !isNaN(parseFloat(n)) && isFinite(n);
+}
+
+module.exports = function (equation) {
+    var ops = equation.split(" ");
+    var result;
+
+    if (ops.length < 2) {
+        result = Number(ops[0]);
+    } else {
+        var curVal = Number(ops[0]);
+        var curOp = ops[1];
+        for (var i = 2; i < ops.length; i++) {
+            if (isNumber(ops[i])) {
+                curVal = mathFuns[curOp](curVal, Number(ops[i]));
```

Comment:
I love the usage of `curOp` as the lookup in `mathFuns`. This will make it simple to support more operations.